### PR TITLE
[REVIEW] Remove copying to host for metadata generation in `generate_pandas_metadata`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #4560

### CODE REVIEW

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

File: python/cudf/cudf/_libxx/parquet.pyx [hunk 1]
```diff
@@ -43,7 +44,7 @@ cpdef generate_pandas_metadata(Table table, index):
     # Columns
     for name, col in table._data.items():
         col_names.append(name)
-        types.append(col.to_arrow().type)
+        types.append(np_to_pa_dtype(col.dtype))
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/parquet.pyx]: Does this handle categorical typed columns as well? If not we'd need it to.

Comment (galipremsagar) [reply python/cudf/cudf/_libxx/parquet.pyx]: Seems like `categorical` types aren't supported yet: https://github.com/rapidsai/cudf/blob/branch-0.13/python/cudf/cudf/io/parquet.py#L191
Added similar validation in meta generation aswell.

**Comment by kkraus14** [comment]
`check-style` errored on launch in CI so it's stuck in pending: https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcudf%2Fprb%2Fcudf-style/detail/cudf-style/6796/pipeline

**Pull request merged**